docs/ops.html

```diff
@@ -2133,6 +2142,136 @@ <h5 class="anchor-heading"><a id="kafka_streams_rocksdb_monitoring" class="ancho
     </tbody>
   </table>
 
+  <strong>RocksDB Properties-based Metrics:</strong>
+  All of the following properties-based metrics have a recording level of <code>info</code> and are recorded when the
+  metrics are accessed.
+  If a state store consists of multiple RocksDB instances, as is the case for aggregations over time and session windows,
+  each metric reports the sum over all the RocksDB instances of the state store, except for the block cache metrics
+  <code>block-cache-*</code>. The block cache metrics report the sum over all RocksDB instances if each instance uses its
+  own block cache, and they report the recorded value from only one instance if a single block cache is shared
```

Comment:
Technically I think it's possible for users to share a single block cache among only some stores, but not others. Or they could have two block caches shared across different state stores, etc. Would we detect this case and report the correct block cache for a given state store?
(Idk how common that pattern could possibly be, but this made me wonder)

Reply:
Good point! Currently, we treat this mixed pattern as an illegal state and throw an `IllegalStateException`. Probably not the best way to handle it. Allowing such a mixed pattern complicates the measurement of the cache metrics. I opened [KAFKA-12223](https://issues.apache.org/jira/browse/KAFKA-12223) to document this.

Reply:
Thanks!